### PR TITLE
fabtests: Add support for FI_MR_RAW keys exchange.

### PR DESCRIPTION
**Pull request by yburette**
Signed-off-by: yohann <yohann.burette@intel.com>

### CODE REVIEW

**Comment by yburette** [comment]
Using the existing tx/rx_buf to exchange the raw keys.
TODO: Ensure that the key fits in those buffers.

File: fabtests/common/shared.c [hunk 1]
```diff
@@ -1240,8 +1285,19 @@ int ft_exchange_keys(struct fi_rma_iov *peer_iov)
...
+			ret = fi_mr_raw_attr(mr, &addr,
+					(uint8_t *) &rma_iov->key, &key_size,
+					0);
+			if (ret)
+				return ret;
+		} else {
+			rma_iov->key = fi_mr_key(mr);
+		}
+
+		ret = ft_tx(ep, remote_fi_addr, len, &tx_ctx);
```
Comment (shefty) [inline fabtests/common/shared.c]: This function is getting too long.  Can you create a new call, ft_exchange_raw_keys(), and move all raw key handling there?  Then we can have a single check for FI_MR_RAW to determine which function to call.

File: fabtests/benchmarks/rma_bw.c [hunk 2]
```diff
@@ -81,6 +81,11 @@ static int run(void)
 			goto out;
 	}
 
+	if (fi->domain_attr->mr_mode & FI_MR_RAW) {
+		ret = fi_mr_unmap_key(domain, remote.key);
+		if (ret)
+			goto out;
+	}
```
Comment (shefty) [inline fabtests/benchmarks/rma_bw.c]: We don't want to update every test file to handle raw keys.  Can this be moved into a common area, similar to how the exchange is done?

**Comment by shefty** [comment]
AWS is reporting this failure:
23:52:24 - name:   fi_dgram_waitset -p "udp"
23:52:24   result: Fail
23:52:24   time:   92
23:52:24   server_cmd: /home/ubuntu/workspace/single-node/PROVIDER/udp/label/ubuntu/fabtests/install/bin/fi_dgram_waitset -p "udp" -s 127.0.0.1
23:52:24   server_stdout: |
23:52:24   client_cmd: /home/ubuntu/workspace/single-node/PROVIDER/udp/label/ubuntu/fabtests/install/bin/fi_dgram_waitset -p "udp" -s 127.0.0.1 127.0.0.1
23:52:24   client_stdout: |
Is there any way to get more information or rerun the test?

**Comment by rajachan** [comment]
Commenting `bot:aws:retest` should retry the test. [This is how we run the single-node test](https://github.com/awslabs/libfabric-ci-scripts/blob/master/single-node.sh#L37). If there are additional flags you'd like to add to runfabtests, we can do that.

cc:@wckzhang

**Pull request merged**